lib/matplotlib/axis.py

```diff
@@ -1012,6 +1013,96 @@ def set_default_intervals(self):
         # attribute, and the derived code below will check for that
         # and use it if it's available (else just use 0..1)
 
+    def _set_lim(self, v0name, v1name, v0, v1, alt0=None, alt1=None, *,
+                 emit=True, auto):
```

Comment:
The signature is too complex and abstracting away the direction makes it even more hard to understand. Since this is internal, we don't have to jump all the hoops of the public set_*lim API. What about:

```suggestion
    def _set_lim(self, v0, v1, vnames=None, *, emit=True, auto):
```
where `v0`, `v1` can optionally be 2-tuples.

This would change the xlim case from:

```
        return self.xaxis._set_lim(
            "left", "right", left, right, xmin, xmax, emit=emit, auto=auto)
```
to
```
        return self.xaxis._set_lim(
            (xmin, left), (ymin, right), lim_names=("left", "right"), emit=emit, auto=auto)
```

Reply:
I could group vnames together, but grouping v0 and alt0 would make the logic in set_lim even worse, because now v0 being a tuple could come either from grouping v0 and alt0 or from v0 being passed in as a (v0, v1) pair...  (unless you want to have the user *always* pass in both v0 and alt0, but that would complicate the implementation of set_inverted.)

Reply:
I see.

What about
```suggestion
    def _set_lim(self, v0, v1, alt0=None, alt1=None, *, vnames=None,
                 emit=True, auto):
```

Reply:
I'm not really sure the first change helps either; e.g. I'll need to immediately unpack `vnames` inside the implementation anyways, and then you're just making the call sites longer if you force an explicit kwarg, even though it's quite clear that `set_lim("left", "right", ...)` is passing argnames.

I can't say I particularly like using e.g. xaxis names here (I don't have a very strong argument against, other than "v0 seems just fine to me"), but I'll let others chime in and can change that if they prefer them too.

Reply:
> it's quite clear that `set_lim("left", "right", ...)` is passing argnames.

I found that very confusing the first time I saw it. I’m coming from set_slim and was expecting two numbers as the first two arguments. The two strings are quite obscure.

I have no strong opinion on x vs. v. Just wanted to mention the possibility.

Reply:
Ah, fair enough.  Renamed to `names`.